### PR TITLE
Update tool recommendation api w transformers

### PR DESCRIPTION
**Pull request by anuprulez**
The PR is from Galaxy main: https://github.com/galaxyproject/galaxy/pull/14791 which updates tool recommendation API according to a new deep learning model based on transformers. 

Bioxiv preprint: https://www.biorxiv.org/content/10.1101/2022.12.16.520746v1

ping @bgruening Thank you!

### CODE REVIEW

File: lib/galaxy/dependencies/conditional-requirements.txt [hunk 1]
```diff
@@ -47,7 +47,7 @@ pygithub
 influxdb
 
 # Deep learning packages for tool recommendation
-tensorflow==2.7.2
+tensorflow==2.9.3
```
Comment (bgruening) [inline lib/galaxy/dependencies/conditional-requirements.txt:50]: Ok, we need to be careful here as the venv also needs to be synced with the celery workers. I would need to talk to Mira how to do this best.

Comment (anuprulez) [reply lib/galaxy/dependencies/conditional-requirements.txt:50]: @bgruening @mira-miracoli how should we proceed here? does Celery get issues with TensorFlow? thanks!

Comment (mira-miracoli) [reply lib/galaxy/dependencies/conditional-requirements.txt:50]: I think we can merge this, it should have no effect on celery

**Comment by anuprulez** [comment]
Thanks @mira-miracoli

**Comment by anuprulez** [comment]
I tried to run this feature on the EU server but am getting `500 error` in the API `get_tool_predictions`

tool-prediction-error-07-02

I started a local Galaxy instance using `release_22.05_europe` branch, and the feature works well. I think on the EU, the correct version of TensorFlow is not installed. Can we check the log to find the exact error? 

local-eu-release


ping @mira-miracoli thanks a lot!

**Comment by mira-miracoli** [comment]
I checked the venv and tensorflow is on version 2.9.3
on sn06 and on celery

**Comment by bgruening** [comment]
@anuprulez can you get me an API request I can try to get you an error log?

**Pull request merged**